### DISCUSSION TITLE
make auto import to support multi root workspace better

### DISCUSSION THREAD

**Discussion by heejaechang** [microsoft/pylance-release#2614] [Enhancement]
multi root workspace with each root being a package and installed as pip install -e ... is a common user scenario but currently auto-import support is very limited.

1. each root in multi workspace is not sharing indices (for third party + other roots)
2. since pip install -e installed package from multi root is treated same as regular third party lib, refresh interval is too slow.
3. it also creates 2 indices for roots. one as user file indices and the other as third party lib to other roots. other root should just use user file indices that contains all symbols defined.

**Comment by heejaechang** [comment]
related issue - https://github.com/microsoft/pylance-release/issues/2914

**Comment by fcabrita** [comment]
Related issue #3152

**Comment by samskiter** [comment]
Is this still an issue? I see a bunch of closed issues but my team to have this still

**Reply by heejaechang** [reply]
can you open a new issue with description on behavior/perf you want or having problem with?
